plugins/destination/plugin_testing_migrate.go

```diff
@@ -74,6 +74,17 @@ func testMigration(ctx context.Context, t *testing.T, p *Plugin, logger zerolog.
 		}
 	}
 
+	if p.spec.PKMode == specs.PKModeCQID {
+		for _, tColumn := range target.Columns {
+			if tColumn.Name != schema.CqIDColumn.Name {
+				continue
+			}
+			if !tColumn.CreationOptions.PrimaryKey {
+				return fmt.Errorf("expected primary key on %s", tColumn.Name)
+			}
+		}
```

Comment:
```suggestion
		col := target.Get(schema.CqIDColumn.Name)
		if col == nil {
			return fmt.Errorf("expected to have %s column", schema.CqIDColumn.Name)
		}
		if !col.CreationOptions.PrimaryKey {
			return fmt.Errorf("expected primary key on %s", col.Name)
		}
```

Reply:
Updated the test to fail if there is a PK on a different column than the cq_id